Evidence fetch (#87)
aragon/court-dashboard: docs/metadata.md

## Dispute

Structure from the graph

```javascript
type Dispute @entity {
  id: ID!
  ...
  subject: Arbitrable!

>   metadata: String!

Reviewer: ```suggestion
  metadata: Bytes!
```

  ...
}
```

```javascript
type Arbitrable @entity {
  id: ID!
  disputes: [Dispute!] @derivedFrom(field: "subject")
}
```

In the frontend we try to parse the `metadata` as a json object.

### metadata - Json object

> We are going to try to find the following keys:

Reviewer: We should update this: if the `metadata` is a json, we now only expect:

```js
1. `description`
2. `metadata`
```

Where `metadata` contains:

```
1. `agreementText` (required)
2. `description`
3. `defendant`
4. `plaintiff`
```

1. `description`
2. `agreementText`
3. `defendant`
4. `plaintiff`

#### metadata - String

> In case that the metadata is just a plain string we are going to use it as the dispute description and the dispute creator `subject.id` will be considered as the plaintiff

Reviewer: ```suggestion
In case `metadata` is a plain string, we assume it to be the dispute's description and set the dispute creator (`subject.id`) as the plaintiff. No agreement link will be available.
```

## Evidence

```javascript
type Evidence @entity {
  id: ID!
  ...

>   data: String!

Reviewer: ```suggestion
  data: Bytes!
```

>   submitter: Bytes!

Reviewer: ```suggestion
  submitter: ID!
```

  ...
}
```

> The dashboard check if the `data` field is a valid IPFS hash/cid

Reviewer: ```suggestion
The dashboard checks if the `data` field is a valid IPFS hash/cid, JSON, or string.
```

### data - IPFS hash

> Try to fetch the content from IPFS

Reviewer: ```suggestion
Try to fetch the content from IPFS and continue with one of the below options.
```

> #### content - Json

Reviewer: Let's throw this case away; having a JSON object may be useful but isn't right now.

We are going to try to find the following keys:

1. `metadata`

If we can find it we are going to show that as the evidence text

#### content - String

Show the content as the evidence text

> ### data - Plain String

Reviewer: Is this the case where `toUtf8()` errors?

In that case we should just say the data is un-parsable.

Author: yeah that's the case where the evidence metadata is not a valid `cid` or does not follow the `ipfs:XXXXX`

Reviewer: In that case, can we rename the section to `data - Unknown String`?
